quasicrystal/operations.py

```diff
@@ -0,0 +1,155 @@
+
+            # generate the pair interaction table
+            cutoff = determineCutoff(potential_k, potential_phi)
+            nl = md.nlist.cell()
+            table = md.pair.table(nlist = nl, width = 1000)
+            table.pair_coeff.set('A', 'A', func = EOPP, rmin = 0.5, rmax = cutoff,
+                                 coeff = dict(k = potential_k, phi = potential_phi))
+
+            # Integrate at constant temperature
+            md.integrate.nvt(group = hoomd.group.all(), tau = 1.0, kT = temperature)
+            md.integrate.mode_standard(dt = 0.01)
+
+            hoomd.run(100000)
+            hoomd.dump.gsd('init.gsd', period=None, overwrite=True,group=hoomd.group.all())
+
+def add_benchmark(project, mode, nranks, gpu_ids=[]):
+    name = ''
+    ngpu = 0
+    if mode == 'cpu':
+        name += 'cpu'
+    elif mode == 'gpu':
+        name += 'gpu'
+        ngpu = max(1,len(gpu_ids))
+        ngpu *= nranks
+    else:
+        raise ValueError('Unknown execution mode')
+
+    name += '_'.join([str(g) for g in gpu_ids])
+    name += '_np{}'.format(nranks)
+
+    @project.operation('{}-benchmark-{}'.format(benchmark_name, name))
+    @project.pre(equilibrated)
+    @project.pre(pertains)
+    @flow.directives(nranks=nranks)
+    @flow.directives(np=nranks)
+    @flow.directives(ngpu=ngpu)
+    def benchmark(job):
+        sp = job.statepoint()
+
+        with job:
+            import hoomd
+            from hoomd import md
+
+            device = hoomd.device.GPU(gpu_ids=gpu_ids) if mode == 'gpu' else hoomd.device.CPU()
+            c = hoomd.context.initialize(device)
+            system = hoomd.init.read_gsd(filename=job.fn('init.gsd'))
+
+            # generate the pair interaction table
+            cutoff = determineCutoff(potential_k, potential_phi)
+            nl = md.nlist.cell()
+            table = md.pair.table(nlist = nl, width = 1000)
+            table.pair_coeff.set('A', 'A', func = EOPP, rmin = 0.5, rmax = cutoff,
+                                 coeff = dict(k = potential_k, phi = potential_phi))
+
+            # Integrate at constant temperature
+            md.integrate.nvt(group = hoomd.group.all(), tau = 1.0, kT = temperature)
+            md.integrate.mode_standard(dt = 0.01)
+
+            nl.set_params(r_buff=0.35, check_period=4)
+
+            # warm up and autotune
+            if c.device.mode == 'gpu':
+                hoomd.run(30000)
+            else:
+                hoomd.run(30000, limit_hours=20.0/3600.0)
+
+            # full benchmark
+            tps = hoomd.benchmark.series(warmup=0, repeat=4, steps=50000, limit_hours=20.0/3600.0)
+            ptps = np.average(tps) * len(system.particles);
+
+            # print out millions of particle time steps per second
+            if c.device.comm.rank == 0:
+                print("Hours to complete 10e6 steps: {0}".format(10e6/(ptps/len(system.particles))/3600));
+                row = dict()
+
+                # meta data
```

Comment:
This is in every benchmark. This might be worthy of a utility function to reduce verbosity and LOC.

Reply:
I am referring to everything below the `# meta data`